bundles/core/src/main/java/com/adobe/cq/commerce/core/cacheinvalidation/internal/InvalidateDispatcherCacheImpl.java

```diff
@@ -39,200 +33,223 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.adobe.cq.commerce.core.cacheinvalidation.internal.spi.CacheInvalidationStrategy;
-import com.adobe.cq.commerce.core.cacheinvalidation.internal.spi.DispatcherCacheInvalidationStrategy;
+import com.adobe.cq.commerce.core.cacheinvalidation.spi.DispatcherCacheInvalidationContext;
+import com.adobe.cq.commerce.core.cacheinvalidation.spi.DispatcherCacheInvalidationStrategy;
+import com.adobe.cq.commerce.core.components.client.MagentoGraphqlClient;
 import com.adobe.cq.commerce.core.components.internal.services.UrlProviderImpl;
 import com.adobe.cq.commerce.core.components.internal.services.site.SiteStructureImpl;
-import com.adobe.cq.commerce.core.components.services.ComponentsConfiguration;
-import com.adobe.cq.commerce.graphql.client.GraphqlClient;
-import com.adobe.cq.commerce.graphql.client.GraphqlRequest;
-import com.adobe.cq.commerce.graphql.client.GraphqlResponse;
 import com.day.cq.wcm.api.Page;
 import com.day.cq.wcm.api.PageManager;
-import com.google.gson.reflect.TypeToken;
 
 @Component(
     service = InvalidateDispatcherCacheImpl.class,
     immediate = true)
 public class InvalidateDispatcherCacheImpl {
 
-    private static final String DISPATCHER_URL = "http://localhost:80/dispatcher/invalidate.cache";
+    private static final String DISPATCHER_BASE_URL = "http://localhost:80";
+    private static final String DISPATCHER_INVALIDATE_PATH = "/dispatcher/invalidate.cache";
+    private static final String PATH_DELIMITER = "/";
+    private static final String IS_FUNCTION = "isFunction";
+    private static final String CQ_ACTION_HEADER = "CQ-Action";
+    private static final String CQ_HANDLE_HEADER = "CQ-Handle";
+    private static final String CQ_ACTION_SCOPE_HEADER = "CQ-Action-Scope";
+    private static final String DELETE_ACTION = "Delete";
+    private static final String RESOURCE_ONLY_SCOPE = "ResourceOnly";
 
     @Reference
     private UrlProviderImpl urlProvider;
 
-    String pathDelimiter = "/";
-
     @Reference
     private SlingSettingsService slingSettingsService;
 
     @Reference
     private InvalidateCacheSupport invalidateCacheSupport;
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(InvalidateDispatcherCacheImpl.class);
-
-    private static final String IS_FUNCTION = "isFunction";
-
     @Reference
     private InvalidateCacheRegistry invalidateCacheRegistry;
 
+    private final HttpClientProvider httpClientProvider = new HttpClientProvider();
+    private static final Logger LOGGER = LoggerFactory.getLogger(InvalidateDispatcherCacheImpl.class);
+
     public void invalidateCache(String path) {
-        if (slingSettingsService.getRunModes().contains("author")) {
-            LOGGER.error("Operation is only supported for author");
+        if (path == null || path.trim().isEmpty() || slingSettingsService.getRunModes().contains("author")) {
+            LOGGER.warn("Invalid path or operation not supported for author");
```

Comment:
This warning is always  logged on author. I think the current method shouldn't be invoked on author instance at all at https://github.com/adobe/aem-core-cif-components/pull/1032/files#diff-56586a55c7c760df5c59832205424a06cb062f76b802b64bd15fad70019aab52R106 .